BiometricLocker/BiometricLocker.swift

```diff
@@ -70,49 +99,95 @@ open class BiometricLocker {
         return false
     }
 
-    /// Locks the app.
-    ///
-    /// - Parameter date: Tells the locker when the app was sent to the background (or any other situation, when applicable), so that we can calculate if the app should be locked.
-    open func lock(at date: Date = Date()) {
+    /// Store the NotificationCenter observers for deallocation.
+    private var notificationObservers = [NSObjectProtocol]()
+
+    public init(localizedReason: String, automaticallyLocksOnBackgroundOrQuit: Bool = true, withUnlockedTimeAllowance: Bool = true) {
+        self.localizedReason = localizedReason
+        
+        if !withUnlockedTimeAllowance {
+            self.unlockedTimeAllowance = 0
+        }
+
+        if automaticallyLocksOnBackgroundOrQuit {
+            self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationDidEnterBackground, object: nil, queue: .main) { _ in
+                // If app goes into background, we start the clocks to lock the app.
+                self.lock()
+            })
+
+            self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationWillTerminate, object: nil, queue: .main) { _ in
+                // If the app is killed, lock it instantly.
+                self.lock(.now)
+            })
+        }
+    }
+
+    deinit {
+        self.notificationObservers.forEach { observer in
+            NotificationCenter.default.removeObserver(observer)
+        }
```

Comment:
It's correct that we need to remove observers on `deinit`. 

What I'm concerned about here is that the `Locker` would get killed by ARC as soon as it goes out of scope unless it's stored as, for instance, a variable on the app delegate (which is, at least in my mind, still a singleton). Then this will fire, the observers will be removed, and it won't actually lock anything.

Reply:
@Elland ^^ this is definitely still a concern